lib/Pass/Pass.cpp

```diff
@@ -133,27 +150,32 @@ struct SouperPass : public FunctionPass {
 
     appendToGlobalCtors(*M, Ctor, 0);
 
+    BasicBlock::iterator BI = I;
+    while (isa<PHINode>(*BI))
+      ++BI;
     new AtomicRMWInst(AtomicRMWInst::Add, CntVar,
                       ConstantInt::get(C, APInt(64, 1)), Monotonic, CrossThread,
                       BI);
   }
 
-  bool runOnFunction(Function &F) {
+  std::set <Function *> DoneFuncs;
+
+  bool runOnFunction(Function *F) {
     bool changed = false;
```

Comment:
Can't remember if there was a naming policy for local vars, shouldn't it be ``Changed``?